producer_test.go

```diff
@@ -77,3 +78,23 @@ func TestProducerMessage(t *testing.T) {
 	listener.Close()
 	producer.Close()
 }
+
+func TestProducerTryToSend(t *testing.T) {
+	producer, _ := new(Producer).InitC("your-stream", "0", "LATEST", "accesskey", "secretkey", "us-east-1", 4)
+	producer.NewEndpoint(testEndpoint, "your-stream")
+	var totDropped int
+	for i := 0; i < 10000; i++ {
+		b := make([]byte, 2)
+		binary.LittleEndian.PutUint16(b, uint16(i))
+		if err := producer.TryToSend(new(Message).Init(b, "foo")); nil != err {
+			totDropped++
+		}
+	}
+	Convey("Given a producer", t, func() {
+		Convey("TryToSend should drop messages when the queue is full", func() {
+			So(totDropped, ShouldBeGreaterThan, 4000)
+			So(totDropped, ShouldBeLessThan, 6000)
```

Comment:
Since producer is a struct and not an interface.  It is difficult to mock out.  However, I can do something else to make the test deterministic.